### PR TITLE
add oci support

### PR DESCRIPTION
**Pull request by fabioDMFerreira**
#577

### CODE REVIEW

File: oci/oci.go [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+	domains, err := identityClient.ListAvailabilityDomains(context.TODO(), identity.ListAvailabilityDomainsRequest{CompartmentId: &p.compartmentID})
+	if err != nil {
+		return
+	}
+
+	if len(domains.Items) > 0 {
+		p.availabilityDomain = *domains.Items[0].Name
+	}
+
+	// instanceID := "ocid1.instance.oc1.us-sanjose-1.anzwuljrltn4jiicpxtrukrjibwzvpxvhjcjvvt3kmxc3civ24vbmv3iskdq"
```
Comment (eyberg) [inline oci/oci.go:135]: can we remove all of this?

File: oci/oci_image.go [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+		return "", err
+	}
+
+	return imagePath, nil
+}
+
+func (p *Provider) createQcow2Image(c *types.Config) (imagePath string, err error) {
+	qcow2ImagesDir, err := findOrCreateqcow2ImagesDir()
+	if err != nil {
+		return
```
Comment (eyberg) [inline oci/oci_image.go:49]: return err?

File: oci/oci_image.go [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+	args := []string{
+		"convert",
+		"-O", "qcow2",
+		c.RunConfig.Imagename, imagePath,
+	}
+
+	cmd := exec.Command("qemu-img", args...)
+	_, err = cmd.CombinedOutput()
+	if err != nil {
+		return
```
Comment (eyberg) [inline oci/oci_image.go:63]: same

File: oci/oci_image.go [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+func (p *Provider) DeleteImage(ctx *lepton.Context, imagename string) (err error) {
+
+	image, err := p.getImageByName(ctx, imagename)
+	if err != nil {
+		return
+	}
+
+	_, err = p.computeClient.DeleteImage(context.TODO(), core.DeleteImageRequest{ImageId: &image.ID})
+
+	return
```
Comment (eyberg) [inline oci/oci_image.go:247]: return err?

**Review by eyberg — APPROVED** [review]

File: oci/oci.go (hunk 1 above)
Comment (fabioDMFerreira) [reply oci/oci.go:135]: Thanks for this. Playground code

File: oci/oci_image.go (hunk 2 above)
Comment (fabioDMFerreira) [reply oci/oci_image.go:49]: Yes. If we initialize the variables on return arguments (`(imagePath string, err error)`) we don't have to specify values on return.

Comment (fabioDMFerreira) [reply oci/oci_image.go:49]: so yes. The error is returned.

Comment (eyberg) [reply oci/oci_image.go:49]: ah i see it's named ok

**Pull request merged**